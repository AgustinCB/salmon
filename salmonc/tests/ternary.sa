print true ? 1 + 2 : 2 - 1;
print false ? 1 + 2 : 2 - 1;
