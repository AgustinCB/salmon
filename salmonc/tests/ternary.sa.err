Constants: [Bool(true), Integer(1), Integer(2), Bool(false), String(0)]
Instructions: [Instruction { instruction_type: Constant(0), location: 0 }, Instruction { instruction_type: JmpIfFalse(4), location: 0 }, Instruction { instruction_type: Constant(1), location: 0 }, Instruction { instruction_type: Constant(2), location: 0 }, Instruction { instruction_type: Plus, location: 0 }, Instruction { instruction_type: Jmp(3), location: 0 }, Instruction { instruction_type: Constant(2), location: 0 }, Instruction { instruction_type: Constant(1), location: 0 }, Instruction { instruction_type: Minus, location: 0 }, Instruction { instruction_type: Constant(3), location: 1 }, Instruction { instruction_type: JmpIfFalse(4), location: 1 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Plus, location: 1 }, Instruction { instruction_type: Jmp(3), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Minus, location: 1 }]
Locations: [Location { address: 0, line: 0 }, Location { address: 0, line: 1 }]
