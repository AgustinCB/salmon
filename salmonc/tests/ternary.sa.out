0 - Integer(1)
1 - Integer(3)
